WalletWasabi.Fluent/ViewModels/AddWallet/WalletNamePageViewModel.cs

```diff
@@ -0,0 +1,117 @@
+using ReactiveUI;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Fluent.Validation;
+using WalletWasabi.Fluent.ViewModels.Dialogs;
+using System.Threading.Tasks;
+using WalletWasabi.Fluent.ViewModels.AddWallet.Create;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.ViewModels.AddWallet.HardwareWallet;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Models;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Fluent.ViewModels.AddWallet;
+
+[NavigationMetaData(Title = "Enter wallet name")]
+public partial class WalletNamePageViewModel : RoutableViewModel
+{
+	[AutoNotify] private string _walletName = "";
+	private readonly string? _importFilePath;
+
+	public WalletNamePageViewModel(WalletCreationOption creationOption, string? importFilePath = null)
+	{
+		EnableBack = true;
+
+		var canExecute = this.WhenAnyValue(x => x.WalletName)
+			.ObserveOn(RxApp.MainThreadScheduler)
+			.Select(x => !string.IsNullOrWhiteSpace(x) && !Validations.Any);
+
+		NextCommand = ReactiveCommand.CreateFromTask(async () => await OnNextAsync(WalletName, creationOption), canExecute);
+
+		this.ValidateProperty(x => x.WalletName, ValidateWalletName);
+		_importFilePath = importFilePath;
+	}
+
+	private async Task OnNextAsync(string walletName, WalletCreationOption creationOption)
+	{
+		switch (creationOption)
+		{
+			case WalletCreationOption.AddNewWallet:
+				await CreatePasswordAsync(walletName);
+				break;
+			case WalletCreationOption.ConnectToHardwareWallet:
+				Navigate().To(new ConnectHardwareWalletViewModel(walletName));
+				break;
+			case WalletCreationOption.RecoverWallet:
+				Navigate().To(new RecoverWalletViewModel(walletName));
+				break;
+			case WalletCreationOption.ImportWallet when _importFilePath is string:
+				await ImportWalletAsync(walletName, _importFilePath);
+				break;
+			default:
+				throw new InvalidOperationException($"WalletCreationOption not supported: {creationOption}");
+		}
+	}
+
+	private async Task ImportWalletAsync(string walletName, string filePath)
+	{
+		try
+		{
+			var keyManager = await ImportWalletHelper.ImportWalletAsync(Services.WalletManager, walletName, filePath);
+			Navigate().To(new AddedWalletPageViewModel(keyManager));
+		}
+		catch(Exception ex)
+		{
+			await ShowErrorAsync("Import wallet", ex.ToUserFriendlyString(), "Error");
```

Comment:
instead of `Error` use something userfriendly message like `Wasabi was unable to import your wallet`.